pkg/logs/internal/decoder/auto_multiline_handler.go

```diff
@@ -152,32 +159,43 @@ func (h *AutoMultilineHandler) processAndTry(message *Message) {
 		h.timeoutTimer = h.clk.Timer(h.matchTimeout)
 	}
 
+	h.linesTested++
+
 	timeout := false
 	select {
 	case <-h.timeoutTimer.C:
 		log.Debug("Multiline auto detect timed out before reaching line test threshold")
+		h.autoMultiLineStatus.SetMessage("message2", fmt.Sprintf("Timeout reached. Processed (%d of %d) logs during detection", h.linesTested, h.linesToAssess))
 		timeout = true
 		break
 	default:
-		break
 	}
 
-	h.linesTested++
+	h.autoMultiLineStatus.SetMessage("state", "State: Using auto multi-line handler")
+	h.autoMultiLineStatus.SetMessage("message", fmt.Sprintf("Detecting (%d of %d)", h.linesTested, h.linesToAssess))
+
 	if h.linesTested >= h.linesToAssess || timeout {
 		topMatch := h.scoredMatches[0]
 		matchRatio := float64(topMatch.score) / float64(h.linesTested)
+		var message string
 
 		if matchRatio >= h.matchThreshold {
-			log.Debugf("Pattern %v matched %d lines with a ratio of %f", topMatch.regexp.String(), topMatch.score, matchRatio)
+			h.autoMultiLineStatus.SetMessage("state", "State: Using multi-line handler")
+			message = fmt.Sprintf("Pattern %v matched %d lines with a ratio of %f", topMatch.regexp.String(), topMatch.score, matchRatio)
```

Comment:
Would it make sense to treat both `state` and `message` the same? Either both set via variables, or with immediate call.